docs/en/stack/security/troubleshooting.asciidoc

```diff
@@ -392,6 +392,123 @@ Kerberos/SPNEGO debug logging on JVM, add following JVM system properties:
+realm configuration or by using the `xpack.security.public.*` settings in `kibana.yml`.
+{kib} either uses the `server.host` and `server.port` or the `xpack.security.public.*`
+settings to construct the SAML ACS URL and this URL needs to match exactly the URL that
+is configured in the SAML realm configuration.
+
+NOTE: The two URLs are compared as case-sensitive strings and not as canonicalized URLs.
+
+. We could not find the metadata for the SAML Entity ID `your:entity.id` in the metadata
+file (`metadata.xml`) that has been configured. Either the value of `idp.entity_id` setting
+in the {es} SAML realm configuration is wrong/misspelled, or the wrong metadata file has been
+configured in `idp.metadata.path`.
+
+. This error indicates that {es} failed to process the incoming SAML authentication message.
+Since the message can't be processed, {es} is not aware of who the to-be authenticated
+user is and the `<unauthenticated-saml-user>` placeholder is used instead. To diagnose the
+_actual_ problem one needs to check the {es} logs for further details.
+
+. This means that the SAML Identity Provider failed to authenticate the user and sent a
+SAML Response to the Service Provider (Elastic Stack) indicating this failure. The `Code`
+can be a different error identifier than the one above and might offer an indication about
+the cause of the failure, but `Message` and `Detail` will be `null`. In order to further
+diagnose the problem please consult the logs and documentation of the Identity Provider.
+
+. This means that {es} failed to validate the digital signature of the SAML message that
+the Identity Provider sent. {es} uses the public key of the Identity Provider that is
+included in the SAML metadata, in order to validate the signature that the IdP has created
+using its corresponding private key. Failure to do so, can have a number of causes:
+.. As the error message indicates, the most common cause is that the wrong metadata file
+is used and as such the public key it contains, doesn't correspond to the private key the
+Identity Provider uses.
+.. The configuration of the Identity Provider has changed or the key has been rotated and
+the metadata file that {es} is using has not been updated.
+.. The SAML Response has been altered in transit and the signature cannot be validated even
+if the correct key is used.
+
+. If you want your users to be able to use local credentials to authenticate to {kib} in
+addition to using the SAML realm for Single Sign-On, you need to enable the `basic`
+`authProvider` in {kib}. The process is documented in the <<saml-kibana-basic, SAML Guide>>
+
+Very detailed trace logging can be enabled specifically for the SAML realm by adding the
+following lines to the end of the `log4j2.properties` configuration file in the `ES_PATH_CONF`:
+
+[source,properties]
+----------------
+logger.saml.name = org.elasticsearch.xpack.security.authc.saml
+logger.saml.level = TRACE
+----------------
```

Comment:
It's really just personal preference, but I recommend adding them to the `elasticsearch.yml` instead, and never touching `log4j.properties`
```
logger.org.elasticsearch.xpack.security.authc.saml: TRACE
```